src/mscorlib/corefx/System/Globalization/TextInfo.Unix.cs

```diff
@@ -61,9 +60,7 @@ private unsafe char ChangeCase(char c, bool toUpper)
         private bool NeedsTurkishCasing(string localeName)
         {
             Contract.Assert(localeName != null);
-
-            string lcName = CultureData.AnsiToLower(localeName);
-            return lcName.Length >= 2 && ((lcName[0] == 't' && lcName[1] == 'r') || (lcName[0] == 'a' && lcName[1] == 'z'));
+            return CultureInfo.GetCultureInfo(localeName).CompareInfo.Compare("i", "I", CompareOptions.IgnoreCase) != 0;
```

Comment:
TextInfos are cached per CultureInfo.  I am hesitant to not do this based on what the collation rules say (e.g. based on the name of the locale) as that could cause a mismatch between collation and casing.